server/src/test/java/org/elasticsearch/search/SearchHitTests.java

```diff
@@ -71,7 +83,7 @@ public static SearchHit createTestItem(boolean withOptionalInnerHits) {
         }
         Map<String, DocumentField> fields = new HashMap<>();
         if (randomBoolean()) {
-            fields = GetResultTests.randomDocumentFields(XContentType.JSON).v1();
+            fields = GetResultTests.randomDocumentFields(xContentType).v2();
```

Comment:
Should this still be `v1`?

Reply:
no, `v2` returns values that will preserve their types when they are (de)serialized via xcontent.
Using `v1` would break the equality since values could change types after serialization (from floats to doubles for instance).

Reply:
Got it. `Tuple` not giving things nice names strikes again.

Reply:
maybe leave a comment here to explain why we get v2 and not v1? I can see this question come up again in the future ;(